### DISCUSSION TITLE
app.tsx 文件配置layout的men，动态菜单到底怎么弄

### DISCUSSION THREAD

**Discussion by liushunli** [umijs/umi#11704] [Q&A] [answered]
![image](https://github.com/umijs/umi/assets/12014406/15cfaac1-dbef-4f02-9417-60293999107a)

app.tsx 文件配置layout的men，动态菜单到底怎么弄，有没有人能帮我具体写一个案例看看，试了半天还是不行

**Comment by fz6m** [comment, marked as answer]
运行时修改路由表需要使用 [`patchClientRoutes`](https://umijs.org/docs/api/runtime-config#patchclientroutes-routes-) ，在 render 请求拿到该用户拥有的路由表配置，之后维护一个路由表的映射，将拥有的路由添加，一个例子可参考 https://github.com/umijs/umi/discussions/9717#discussioncomment-4078053 。

**Reply by liushunli** [reply]
谢谢解决了，动态路由配置成功了
